modules/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/actions/data/sql/client/AbstractSQLAction.java

```diff
@@ -774,4 +775,32 @@ private BDataTable constructDataTable(ResultSet rs, Statement stmt, Connection c
         ArrayList<ColumnDefinition> columnDefinitions = getColumnDefinitions(rs);
         return new BDataTable(new SQLDataIterator(conn, stmt, rs, utcCalendar, columnDefinitions));
     }
+
+    private String getSQLType(BStruct parameter) {
+        String sqlType = "";
+        BEnumerator typeEnum = (BEnumerator) parameter.getRefField(0);
+        if (typeEnum != null) {
+            sqlType = typeEnum.getName();
+        }
+        return sqlType;
+
+    }
+
+    private int getParameterDirection(BStruct parameter) {
+        int direction = 0;
```

Comment:
What is "0"?, you've to use the constant to initialize this. Also, shall we use a Java Enum for the direction values here, rather than using an int.